pkg/minikube/download/preload.go

```diff
@@ -169,6 +178,24 @@ func Preload(k8sVersion, containerRuntime, driverName string) error {
 		return err
 	}
 
+	if driverName == driver.HyperV {
+		u, err := url.Parse(windowsServerIsoURL)
+		if err != nil {
+			return errors.Wrapf(err, "url.parse %q", windowsServerIsoURL)
+		}
+
+		// It's already downloaded
+		if u.Scheme == fileScheme {
+			return nil
+		}
+
+		out.Step(style.FileDownload, "Downloading Windows Server 2022 ISO ...")
+
+		if err := downloadWindowsISO(windowsServerIsoURL, WindowsServerPath()); err != nil {
+			return errors.Wrapf(err, "download failed: %s", windowsServerIsoURL)
+		}
+	}
+
```

Comment:
1. This shouldn't go in the `preload.go` file, preloads are specific to caching K8s images and binaries. This should go in the `iso.go` file in the same folder.

2. We shouldn't download the Windows Server ISO for users until they request it, even if 25% of users used the Windows ISO, we're still unnecessarily downloading for the other 75%.

Reply:
The current logic change addresses this 

1. If the user requests for the ISO then we download and cache it
2. If the user starts a cluster with windows node expected then we check if the ISO exists if not we download it
3. I'll also add this to the node add command for a windows node. We will first check if the ISO exists if not then we download it!

Reply:
I think we should only download the windows iso if a user has specified they want a windows node. Else minikube start should not download it by default